pkg/common/common.go

```diff
@@ -239,3 +241,57 @@ func GetImagePullPolicy(image string) (corev1.PullPolicy, error) {
 	}
 	return corev1.PullAlways, nil
 }
+
+// GenerateCliArgsFromConfigMap generates CLI arguments from a ConfigMap.
+//
+// This function takes a ConfigMap and a CLI subcommand(s), and returns a slice of strings representing
+// the CLI arguments from the subcommand(s) followed by the arguments from the ConfigMap.
+// The function processes each key-value pair in the ConfigMap as follows:
+//
+//  1. If the ConfigMaps' value starts and ends with single (') or double (") quotes, it is left unchanged.
+//  2. If the ConfigMaps' key starts with single '-' or double '--', it is left unchanged.
+//  3. If an argument key is passed without `-` or `--` prefix, then `--` is added as a key prefix.
+//  4. If the ConfigMap value is "true" or "false" (case-insensitive), it is converted to lowercase
+//     and used without single quotes surroundings (boolean value).
+//  5. The formatted key-value pair is added to the result that is alphabetically sorted.
+//
+// Args:
+//
+//		configMap: A pointer to a corev1.ConfigMap containing key-value pairs.
+//		cliSubCommand: The CLI subcommand(s) as a string, for example 'server'
+//	                or 'node-agent', 'server'
+//
+// Returns:
+//
+//	A slice of strings representing the CLI arguments.
+func GenerateCliArgsFromConfigMap(configMap *corev1.ConfigMap, cliSubCommand ...string) []string {
```

Comment:
would remove `cliSubCommand` arg from function to make it simpler

Reply:
how would you make it generic then?

We want to pass part of the args from the reconcile and second part from the configmap

Reply:
just append return from this function to an array that already contains the sub commands when using it in controllers

Reply:
I prefer this command to contain all args and sub-args, so we don't have mistakenly two implementations of the cliSubCommands within two different reconcile loops.